### ISSUE TITLE
Replacement of hardwarecfg with mhwdcfg

### ISSUE THREAD

**Issue by hphilm** [calamares/calamares-manjaro#24]
Before we replace **hardwarecfg** we have to review the new code of **mhwdcfg** and think on how we handle following things:
- [ ] Remove virtualbox driver on real hardware, as some editions ship it preinstalled. There is no option for it yet
- [ ] Check if we still need configuration of ALSA. If so, we should move it to **postcfg**
- [ ] Check if we still need configuration of PULSE. If so, we should move it to **postcfg**

**Comment by udeved** [comment]
Right, mhwdcfg work if local is true, ie install from mounted mhwd repo in chroot.
I am currently fixing mhwdcfg if local false, ie online packages, tricky if chrootcfg is used, currently also addressing it. Either networkcfg module doesn't properly work, or the africa mirror is absolutely out of sync and terrible.

Btw, virtualbox drivers, shouldn't be necessary to remove them, since no present in root or DE image.

**Comment by hphilm** [comment]
Well, for the network thing with the mirrors we should deal with it properly by configuring the mirrors by country. Checkout the new code of **locales** module. Teo added some **geoip** detection code to it. Anyway, that selection of country should be used to generate the mirrorlist. If the country is not found as we might have no mirror for it yet, we should autoconfig the mirrorlist.

**Comment by udeved** [comment]
I had the same idea in mind, to use lang selection to match mirror location.
For the moment, ranking code is in chrootcfg, but we could start a manjaro specific code module that deals with mirrors. However, such module would ideally have a branch selector.

**Comment by udeved** [comment]
@philmmanjaro 

I merged latest changes and fixes
mhwdcfg is now ready for testing

One thing, if mhwd is configured for local repo, and netinstall only installs base group, or unpackfs only extracts root-image, mhwd fails, since not all required packages are in the local mhwd repo.

**Comment by hphilm** [comment]
Well, we have to see about that. Currently building some kernels ...

**Comment by udeved** [comment]
I am about to upload cal-dev so we can test it all.
Regarding mhwd repo, its basically more a netinstall issue, since I can't add logic to select or deselect particular groups, if some other group is selected

**Comment by hphilm** [comment]
Ping me if **mt-0.13.4** is ready to be released ...

**Comment by udeved** [comment]
Should be today or tomorrow, but my previous tests were all fine.
I guess we can do it today.

**Comment by udeved** [comment]
I don't think we need pulse and alsa stuff.
It all works fine in live session without it so far.

We don't need to remove vbox driver, since there is none installed in target root.

**Comment by hphilm** [comment]
Right, however let me double check that with the virtualbox drivers.

**Comment by udeved** [comment]
We run mhwd in chroot, and image is clean.

**Comment by hphilm** [comment]
Ok, lets try it then with the next **calamares-dev** build ...

**Comment by udeved** [comment]
mt-0.13.4 already use it.

**Comment by hphilm** [comment]
Well, **calamares** package don't ship **mhwdcfg** module at all. So we have to test it first and add it later to the stable package when ready.

**Comment by udeved** [comment]
I have set cal-dev in profiles for now.

**Comment by hphilm** [comment]
I assume in **master** branch. I'm aiming for the **16.08-preX** branch which is still based on **mt-0.12.x** series. However go on with your development.

**Comment by udeved** [comment]
just saying, shared packages-live has cal-dev set. So temporarily override it, or else we get tons of complaints from team that cal doesn't work. ;)

**Comment by hphilm** [comment]
Sure. Works for me as long as these packages stay in **unstable** for now. It is great that the team is working so hard to finally getting **netinstall** working. Was needed for a long time.

**Comment by udeved** [comment]
We only need to establish how we deal with the netgroups, per profile?, one fits all?, should we have a yaml repo? Could netinstall module accept local yaml netgroups...? Tons of questions. :)
Online netgroups is cool however, you can test them in one running session.

**Comment by hphilm** [comment]
After I found some time to test the new **calamares** package, I might also find some time to remove **hardwarecfg** in **calamares-dev** and test it also. Good work on the module!

**Comment by udeved** [comment]
We need to properly test chrootcfg(for net boot strap), mhwdcfg, servicescfg(openrc),
But they should all work.

**Comment by hphilm** [comment]
Sure, but lets do it one by one and first with regular installation procedure, which is using unsquash.

**Comment by udeved** [comment]
This one worked in quite a few test installs. ;)

**Comment by hphilm** [comment]
Can someone point me to the exact **commit** so I can review these changes? I assume we are all talking about this now:

To automatically set `QT_QPA_PLATFORMTHEME` for user session, add the following line to `~/.profile`:

```
[ "$XDG_CURRENT_DESKTOP" = "KDE" ] || [ "$XDG_CURRENT_DESKTOP" = "GNOME" ] || export QT_QPA_PLATFORMTHEME="qt5ct"
```

**Comment by hphilm** [comment]
We also have to check on how we fix it for **root** account, as **Manjaro-Settings-Manager** has theme issues using **pkgexec** otherwise. **kdesu** and **sudo -E** seems to use the user session settings just fine.
